### ISSUE TITLE
Unable to get basic pages project with worker working

### ISSUE THREAD

**Issue by johanndev** [cloudflare/wrangler2#50]
### System information
```
Windows 10 Version 21H2 (Build 19044.1348)
node v16.13.0

EDIT: also tested on Ubuntu 18.04.6 LTS via WSL, same behaviour
```

Hi! 

I'm really excited to try the recently announced fullstack pages experience, unfortunately I'm unable to get a basic example working.

## Steps to reproduce:
1. In an empty directory, initialize a new project: `npm init -y`
2. Install wrangler2: `npm i -D wrangler@beta` (version resolved as: `"^0.0.0-fsw-beta.5"`)
3. Create `index.html` in the `./static` directory:
```html
<!DOCTYPE html>
<head>
    <script type="module">
        const button = document.getElementById("button")
        button.onclick = () => {
            fetch("/api/foo", {
                method: "GET"
            })
        }
    </script>
</head>
<body>
    <button type="button" id="button">Call function</button>
</body>
</html>
```

4. Create `_worker.js` in the `./static` directory:
```javascript
export default {
    async fetch(request, env) {
      const url = new URL(request.url);
      if (url.pathname.startsWith('/api/')) {
        // TODO: Add your custom /api/* logic here.
        return new Response('Ok');
      }
      // Otherwise, serve the static assets.
      // Without this, the Worker will error and no assets will be served.
      return env.ASSETS.fetch(request);
    }
}
```
5. Start project: `npx wrangler@beta pages dev ./static --local`:
```bash
Serving ./static...
Sleeping 5 seconds to allow proxy process to start before attempting to automatically determine port...
To skip, specify the proxy port with --proxy.
[proxy]: npm
[proxy]:  WARN exec The following package was not found and will be installed: serve

No Instance(s) Available.
Could not automatically determin proxy port. Please specify the proxy port with --proxy.
[proxy]: INFO: Accepting connections at http://localhost:3000
```
6. Navigating to `http://localhost:3000` correctly serves the index page, clicking the button results in a `404`, with no output from the wrangler CLI.

### Expected result
The call to `/api/foo` is handled by the worker defined in `_worker.js`
### Actual result
The call to `/api/foo` results in a `404`

I'm pretty sure I'm overlooking something really basic :)

**Comment by johanndev** [comment]
Figured it out - when serving static files, the `--port` switch is _not_ optional; i.e., this works:
```
npx wrangler@beta pages dev --proxy 3000 --local ./static
Serving ./static...
[HPM] Proxy created: /  -> http://127.0.0.1:59879
[HPM] Proxy created: /  -> http://127.0.0.1:3000
Serving at http://127.0.0.1:8788/
[proxy]: INFO: Accepting connections at http://localhost:3000
```
The site is now reachable under `http://127.0.0.1:8788/` and the call to the function works! :)

Is this expected behaviour?

**Comment by GregBrimble** [comment]
Hi!

The `--proxy` flag should be optional, so if you're needing to specify it, that's a bug. But, I'm not able to reproduce on macOS with that version of wrangler. Can I ask what flavor of machine you're using?

This is what I see (note it takes the `Automatically determined the proxy port to be 3000.` and subsequent messages appear after five seconds):
```
➜  static-test npx wrangler pages dev ./dist
Serving ./dist...
Sleeping 5 seconds to allow proxy process to start before attempting to automatically determine port...
To skip, specify the proxy port with --proxy.
[proxy]: INFO: Accepting connections at http://localhost:3000

Automatically determined the proxy port to be 3000.
[HPM] Proxy created: /  -> http://127.0.0.1:64359
[HPM] Proxy created: /  -> http://127.0.0.1:3000
Serving at http://127.0.0.1:8788/
```

We're working on replacing the current static directory server stuff this week, so this might just coincidentally get patched up anyway, but thanks for the report.

**Comment by johanndev** [comment]
> Can I ask what flavor of machine you're using?

```
Windows 10 Version 21H2 (Build 19044.1348)
node v16.13.0

EDIT: also tested on Ubuntu 18.04.6 LTS via WSL, same behaviour
```

> We're working on replacing the current static directory server stuff this week

I will try to run my sample again after the new static file server lands :)

**Comment by GregBrimble** [comment]
We've just added the new static server to the latest alpha. Can you try that out (`npm install wrangler@alpha`) and see if you're still experiencing this issue? Thanks!

**Comment by johanndev** [comment]
Using the setup from above, I now get (with version `0.0.0-23543fe`):
```
➜ npx wrangler@alpha pages dev --local ./static
Serving at http://127.0.0.1:8788/
Could not serve static asset: TypeError: Failed to parse URL from [object Object]
GET /: FetchError [ERR_RESPONSE_TYPE]: Fetch handler didn't respond with a Response object.
Make sure you're returning a `Response` in your handler.
    at EventTarget.[kDispatchFetch] (file:///C:/Users/<username>/AppData/Local/npm-cache/_npx/c2aa6e5773d6d3ef/node_modules/@miniflare/core/dist/src/index.js:1140:15)
    at processTicksAndRejections (node:internal/process/task_queues:96:5)
    at async Server.<anonymous> (file:///C:/Users/<username>/AppData/Local/npm-cache/_npx/c2aa6e5773d6d3ef/node_modules/@miniflare/http-server/dist/src/index.js:443:20)
Could not serve static asset: TypeError: Failed to parse URL from [object Object]
GET /favicon.ico: FetchError [ERR_RESPONSE_TYPE]: Fetch handler didn't respond with a Response object.
Make sure you're returning a `Response` in your handler.
    at EventTarget.[kDispatchFetch] (file:///C:/Users/<username>/AppData/Local/npm-cache/_npx/c2aa6e5773d6d3ef/node_modules/@miniflare/core/dist/src/index.js:1140:15)
    at processTicksAndRejections (node:internal/process/task_queues:96:5)
    at async Server.<anonymous> (file:///C:/Users/<username>/AppData/Local/npm-cache/_npx/c2aa6e5773d6d3ef/node_modules/@miniflare/http-server/dist/src/index.js:443:20)
```

**Comment by elemental-mind** [comment]
I run into similar issues. Tried the beta and the alpha version of Wrangler - but I can't get functions to work AND I also manually need to specify `--proxy 3000` in order for everything to properly start up. Running Win 10 Version 10.0.19043.

But I have found the culprit: You have a bug in your code generation!

I have a `functions` folder in my top level project and a `functions/api/helloworld.js` that echoes a simple "Hello World".

```
C:\...\Desktop\cloudflare-pages>npx wrangler@alpha pages dev --local --proxy 3000 ./ 
Serving ./...
[proxy]: INFO: Accepting connections at http://localhost:3000

Compiling worker to "C:\\...\\AppData\\Local\\Temp\\functionsWorker.js"
Finished in 56ms.
[HPM] Proxy created: /  -> http://127.0.0.1:56206
[HPM] Proxy created: /  -> http://127.0.0.1:3000
Serving at http://127.0.0.1:8788/
```

Calling  `http://127.0.0.1:8788/api/helloworld` yields a `404`. Upon closer inspection of the generated js file `C:\\...\\AppData\\Local\\Temp\\functionsWorker.js`, though I found out that your `routes` object is defined as follows:

```
var routes = [
  {
    routePath: "/apihelloworld",
    methods: [],
    middlewares: [],
    modules: [onRequest]
  }
];
```
`http://127.0.0.1:8788/apihelloworld` indeed yields the expected "Hello World". So just add a slash and things should be working already better.

**Comment by GiancarlosIO** [comment]
Same error here using win10 😢 
In my case, it's generating this for a /functions/api/[[path]].js
![image](https://user-images.githubusercontent.com/17274950/147031520-f24735d1-fc83-4312-a908-83c28d9b51bd.png)

**Comment by Mexican-Man** [comment]
It might be a little more complicated. My route: `/v1/tokens/` got turned into `/v1 okens`. I have other routes with "t" in them that are fine, maybe something to do with "/t" in particular.

**Comment by GregBrimble** [comment]
This should be fixed with #331 . Please try it out and let us know if you have any other problems running on Windows!
